automation/jleechanorg_pr_automation/automation_utils.py

```
@@ -0,0 +1,312 @@
+        """Safely read JSON file with file locking"""
+        try:
+            with open(file_path) as f:
+                fcntl.flock(f.fileno(), fcntl.LOCK_SH)  # Shared lock for reading
+                data = json.load(f)
+                fcntl.flock(f.fileno(), fcntl.LOCK_UN)  # Unlock
+                return data
+        except (FileNotFoundError, json.JSONDecodeError):
+            return {}
+
+    @classmethod
+    def safe_write_json(cls, file_path: Path, data: dict):
```

Ruff (0.14.5)

Missing return type annotation for classmethod `safe_write_json`

Add return type annotation: `None`

(ANN206)